flox-bash/tests/integration.bats

```diff
@@ -229,24 +229,24 @@ load test_support.bash
 @test "flox list after install should contain hello" {
   run $FLOX_CLI list -e $TEST_ENVIRONMENT
   assert_success
-  assert_output --partial "Curr Gen  2"
-  assert_output --regexp "0  stable.nixpkgs-flox.hello +"$VERSION_REGEX
+  assert_output --regexp "Curr Gen +[0-9]+"
+  assert_output --regexp "[0-9]+ +stable.nixpkgs-flox.hello +$VERSION_REGEX"
```

Comment:
I restored the hard coded indexes and modified the versions appropriately.

I've also moved out tests associated with `etc-profile` and `FLOX_ENV` to separate files to avoid interspersing them among tests which are sensitive to generation numbers and such.